service/orchestrator/toe.go

```diff
@@ -382,31 +380,20 @@ func getControls(controls []*orchestrator.Control, levels []string, level string
 		return c, err
 	}
 
-	// Add controls based on their assurance level to the lists low, medium and high. If a controls is not defined regarding the assurance level it is dropped.
-	for i := range controls {
-		switch controls[i].GetAssuranceLevel() {
-		case levels[0]:
-			low = append(low, controls[i])
-		case levels[1]:
-			medium = append(medium, controls[i])
-		case levels[2]:
-			high = append(high, controls[i])
-		default:
+	// Add controls based on their assurance level to the map. If a controls is not defined regarding the assurance level it is dropped.
+	for i, control := range controls {
+		if control.AssuranceLevel == nil {
 			continue
 		}
+		levelControls[controls[i].GetAssuranceLevel()] = append(levelControls[controls[i].GetAssuranceLevel()], control)
```

Comment:
why are you using `controls[I]` here and `control` in line 385. Since `control` is a pointer, you can just drop the index and use it in both occasions.

Reply:
:+1: I actually wanted to write the following, but deepSource does not accept it and then forgot to delete the i again: `levelControls[controls[i].GetAssuranceLevel()] = append(levelControls[controls[i].GetAssuranceLevel()], controls[i])`

Reply:
Is deep source accepting this?

```
levelControls[control.GetAssuranceLevel()] = append(levelControls[control.GetAssuranceLevel()], control)
```

Reply:
Or maybe its cleaner to write
```go
level := control.GetAssuranceLevel()
levelControls[level] = append(levelControls[level], control)
```